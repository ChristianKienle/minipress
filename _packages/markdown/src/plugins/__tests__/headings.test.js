// @ts-check
const createMd = require('./../../create-md')
const headingsPlugin = require('./../headings')

describe('headings plugin', () => {
  it('works', () => {
    const md = createMd().use(headingsPlugin)
    const page = { headings: [] }
    const env = {
      page
    }
    expect(md.render(`
# Title
Text

## Chapter 1
## Chapter 2
`, env)).toMatchSnapshot()
const { headings } = page
    expect(headings).toHaveLength(3)
    expect(headings[0]).toMatchObject({ text: 'Title', level: 1, slug: 'title'})
    expect(headings[1]).toMatchObject({ text: 'Chapter 1', level: 2, slug: 'chapter-1'})
    expect(headings[2]).toMatchObject({ text: 'Chapter 2', level: 2, slug: 'chapter-2'})
  })
})