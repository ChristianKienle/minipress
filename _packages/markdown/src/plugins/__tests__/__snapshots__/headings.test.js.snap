// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`headings plugin works 1`] = `
"<h1 id=\\"title\\"><router-link class=\\"header-anchor\\" to=\\"#title\\" aria-hidden=\\"true\\">#</router-link> Title</h1>
<p>Text</p>
<h2 id=\\"chapter-1\\"><router-link class=\\"header-anchor\\" to=\\"#chapter-1\\" aria-hidden=\\"true\\">#</router-link> Chapter 1</h2>
<h2 id=\\"chapter-2\\"><router-link class=\\"header-anchor\\" to=\\"#chapter-2\\" aria-hidden=\\"true\\">#</router-link> Chapter 2</h2>
"
`;
